### PR TITLE
Lockwise keychain launch issues

### LINKED ISSUE #1190: Lockwise is unusable after following the STR
## Steps to reproduce
1. Login to FxA
2. Disconnect from FxA
3. Login to FxA
4. Close Lockwise from App Switcher
5. Re-open Lockwise 

### Actual behavior
- Lockwise is unusable.

### Device & build information
* Device: iPhone 7
* Build version: 1.7.2 (4059)
### Notes
- Possibly a regression from #1181
Attachments: 
[Lockwise unusable.mov.zip](https://github.com/mozilla-lockwise/lockwise-ios/files/4171212/Lockwise.unusable.mov.zip)

### PR DESCRIPTION
**Pull request by kaylagalway**
Fixes #1190 

- Change was made in [PR 1178](https://github.com/mozilla-lockwise/lockwise-ios/pull/1178) that erased entire Keychain instead of specific values when a user logged out. This PR makes a change that specifies when to erase user sync account values and local database values.  This resolves lockwise being unusable after logging out and back in.

To test:
- Launch lockwise with fresh install
- Log in to sync account, verify sync is successful
- Log out of sync account
- Log in to sync account, verify sync is successful
- Force quit app
- Relaunch app and app should continue to sync normally

### CODE REVIEW

File: Shared/Store/BaseDataStore.swift [hunk 1]
```diff
@@ -109,7 +109,7 @@ class BaseDataStore {
     }()
 
     internal var loginsKey: String? {
-        let key = "sqlcipher.key.logins.db"
+        let key = KeychainKey.loginsKey.rawValue
```
Comment (jhugman) [inline Shared/Store/BaseDataStore.swift]: 😸

File: lockbox-ios/Common/AppDelegate.swift [hunk 2]
```diff
@@ -25,7 +25,7 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     func application(_ application: UIApplication, willFinishLaunchingWithOptions
                      launchOptions: [UIApplication.LaunchOptionsKey: Any]? = nil) -> Bool {
         if isFirstRun {
-            KeychainWrapper.wipeKeychain()
+            KeychainWrapper.sharedAppContainerKeychain.clearAllValues(for: .account)
```
Comment (jhugman) [inline lockbox-ios/Common/AppDelegate.swift]: IIUIC, we should be able to clear values for `.database` here as well.

File: lockbox-ios/Store/AccountStore.swift [hunk 3]
```diff
@@ -142,7 +142,7 @@ extension AccountStore {
     }
 
     private func clear() {
-        KeychainWrapper.wipeKeychain()
+        keychainWrapper.clearAllValues(for: .account)
```
Comment (jhugman) [inline lockbox-ios/Store/AccountStore.swift]: IIUC This is the point which fixes the bug.

File: lockbox-ios/Common/AppDelegate.swift [hunk 4]
```diff
@@ -107,7 +107,8 @@ extension AppDelegate {
         } else if previous > current {
             //this would mean a user had an upgraded test version and has now downgraded to a previous version
             //we should wipe keychain data to remove any possible abnormalities
-            KeychainWrapper.wipeKeychain()
+            KeychainWrapper.sharedAppContainerKeychain.clearAllValues(for: .account)
+            KeychainWrapper.sharedAppContainerKeychain.clearAllValues(for: .database)
```
Comment (jhugman) [inline lockbox-ios/Common/AppDelegate.swift]: In time, will we move back to `wipeKeychain()`?

**Comment by kaylagalway** [comment]
> I very much like the approach you have here.
> 
> In the cases where you are wiping both `.account` and `.database`, I would recommend either an `.all` case, or just wiping the keychain as before.
> 
> Excellent work.

Agreed - I will add a `.all` case!

File: lockbox-ios/Store/AccountStore.swift (hunk 3 above)
Comment (kaylagalway) [reply lockbox-ios/Store/AccountStore.swift]: Yes 👍

File: lockbox-ios/Common/AppDelegate.swift (hunk 4 above)
Comment (kaylagalway) [reply lockbox-ios/Common/AppDelegate.swift]: Yes. I like being explicit about the values for now, but I think in the future we could do this.

File: lockbox-ios/Common/AppDelegate.swift (hunk 2 above)
Comment (kaylagalway) [reply lockbox-ios/Common/AppDelegate.swift]: Agreed

**Pull request merged**